### PR TITLE
Space Travelers hub App 

### PR DESCRIPTION
**Pull request by ridabensalem**
In this milestone of the Space travelers hub we have added the following requirements : 

- Create React App.
- Download the Logo Image  
- Create routes and view components: rockets, missions, my profile. Add dragons.
- Use <NavLink /> for the page navigation links and style active class to indicate which section/page the user is currently on.
- Create directories for all Redux state slice files: rockets, missions, dragons.
- Redux: Fetch data and update the Redux store
- Upon the first render fetch data from the SpaceX API endpoints:
1. Rockets: https://api.spacexdata.com/v4/rockets
2. Missions: https://api.spacexdata.com/v3/missions
3. Dragons: https://api.spacexdata.com/v4/dragons 

- Once the data are fetched, dispatch an action to store the selected data in the Redux store:
- [ ] Rockets:
- [ ] id
- [ ] rocket_name
- [ ] description
- [ ] flickr_images
- [ ] Missions:
- [ ] mission_id
- [ ] mission_name
- [ ] description
- [ ] Dragons [only if your team has 3 members]:
- [ ] id
- [ ] name
- [ ] type
- [ ] flickr_images NOTE

- Render UI: lists
- Use useSelector() Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes. 

- Styling the app in CSS
- Render a list of rockets (as per design).
-  For the image of a rocket using the first image in the array of flickr_images.

- Render a table with the missions' data (as per design).
- Render a list of dragons (as per design). For the image of a dragon use the first image in the array of flickr_images.

- Redux: Write actions and reducers for booking rockets/dragons and joining missions
- When a user clicks the "Reserve rocket" button or "Reserve dragon" button, action needs to be dispatched to update the store. 
- Create a reducer and action dispatcher for the "Join Mission" button. 
- Redux: Write actions and reducers for canceling rockets/dragons and leaving missions
- Here you need to follow the same logic as with the "Reserve rocket"/"Reserve dragon" and "Join mission" - but you need to set the reserved key to false.
- Dispatch these actions upon clicking on the corresponding buttons.
- Render UI: conditional components rendering
- Rockets that have already been reserved should show a "Reserved" badge and a "Cancel reservation" button instead of the default "Reserve rocket" (as per design) .
Dragons that have already been reserved should show a "Reserved" badge and a "Cancel reservation" button instead of the default "Reserve dragon".
- Missions that the user has joined already should show a badge "Active Member" instead of the default "NOT A MEMBER" and a button "Leave Mission" instead of the "Join Mission" button (as per design).
- - Rockets/Dragons and Missions should use the React conditional rendering syntax:
- - Render UI: My Profile section
- - Compose a two/three-column layout and list ONLY the rockets/dragons reserved and missions joined by the user (as per design):
Render a list of all joined missions (use filter()).
Render a list of all reserved rockets (use filter()).
Render a list of all reserved dragons (use filter()) .

### CODE REVIEW

File: src/features/missionSclice.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+//         state.missionStore = action.payload;
+//       })
+//       .addCase(fetchMission.rejected, (state, action) => {
+//         state.status = 'failed';
+//         state.error = action.error.message;
+//       });
+//   },
+// });
+
+// export default missionSlice.reducer;
```
Comment (m05e5) [inline src/features/missionSclice.js:75]: - [ ] Please always make sure to keep your codebase clean and organized. kindly remove all unnecessary comments

File: src/components/dragons.jsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  image: PropTypes.string.isRequired,
+  reserved: PropTypes.bool.isRequired,
+};
+
+function Dragons() {
+  const dragonsArr = useSelector((state) => state.dragonReducer.dragonStore);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchDragon());
+  }, [dispatch]);
```
Comment (m05e5) [inline src/components/dragons.jsx:64]: - [ ] You really did an amazing job. unlike the rockets, every time I make dragon reservations, navigate and come back to the dragon page, the state is re-initialized and all my reservations are wiped out please kindly fix it

File: src/components/missions.jsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  name: PropTypes.string.isRequired,
+  description: PropTypes.string.isRequired,
+  reserved: PropTypes.string.isRequired,
+};
+function Missions() {
+  const missionsArr = useSelector((state) => state.missionReducer.missionStore);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchMission());
+  }, [dispatch]);
```
Comment (m05e5) [inline src/components/missions.jsx:81]: - [ ] You really did an amazing job. unlike the rockets, every time I join missions, navigate and come back to the mission page, the state is re-initialized and all my reservations are wiped out please kindly fix it

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far. !
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights
- Very beautiful UI
- no linters error
- tests working properly

## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [ ] Please check the comments below 

_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi team 👋🏽 

To highlight 👍🏽 
✔️ License is fixed
✔️ No commented code
✔️ Dragons and missions are successfully reserved

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**